bamToFastq.wdl

```diff
@@ -0,0 +1,341 @@
+version 1.0
+
+workflow bamToFastq {
+    input {
+        File bamFile
+        String fileNaming
+    }
+
+    parameter_meta {
+        bamFile: "A BAM file with one or more readgroups"
+        fileNaming: "The naming scheme for the extracted FASTQs"
+
+    }
+
+    meta {
+        author: "Murto Hilali"
+        email: "mhilali@oicr.on.ca"
+        description: "Generating R1 and R2 from bam files"
+        dependencies: [
+            {
+                name: "java/8",
+                url: "https://www.java.com/en/download/manual.jsp"
+            },
+            {
+                name: "samtools/1.9",
+                url: "https://github.com/samtools/samtools/archive/0.1.19.tar.gz"
+            },
+            {
+                name: "picard/2.21.2",
+                url: "https://broadinstitute.github.io/picard/command-line-overview.html"
+            },
+            {
+                name: "python/3.6",
+                url: "https://www.python.org/downloads/"
+            }
+        ]
+        output_meta: {
+            bamFileFlagstat: "A TXT file containing flag information about the BAM file",
+            readGroups: "A TXT file containing information about the merged BAM file"
+        }
+    }
+
+    call countFlags {
+        input:
+            bamFile = bamFile
+    }
+
+    call nameCheck {
+        input:
+            readGroups = countFlags.readGroups,
+            fileName = fileNaming
+    }
+
+    if (nameCheck.valid == true){
+        call backExtract { 
+            input:
+                bamFile = bamFile
+        }
+    }
+
+    call renameFastqs { 
```

Comment:
Maybe move this under #54 if (nameCheck.valid == true) ? the task backExtract is called only if nameCheck is successfull